packages/compass-maybe-protect-connection-string/src/index.ts

```diff
@@ -0,0 +1,9 @@
+import preferences from 'compass-preferences-model';
+import { redactConnectionString } from 'mongodb-connection-string-url';
+
+export function maybeProtectConnectionString(connectionString: string): string {
+  if (preferences.getPreferences().protectConnectionStrings) {
+    return redactConnectionString(connectionString);
+  }
+  return connectionString;
+}
```

Comment:
Obviously this is a bunch of overhead for a single function, but at the same time: :woman_shrugging: ?

This might have fit into compass-utils but that would have resulted in a circular dependency, and it’s not like it really hurts to have another package with our current monorepo setup.